NH-3958: documents missing types, fix broken links

fredericDelaporte opened the pull request:
[NH-3958](https://nhibernate.jira.com/browse/NH-3958): adding some missing NHibernate types in reference documentation.
Fix two broken links, one updated, the other one removed (referencing old Nullables library).
Pushed api version in documentation from 4.0 to 5.0.

gliljas commented:
Nice! But I see you've also added a Sandcastle project file. Please remove that.

hazzik commented on doc/reference/modules/nhibernate_mapping_attributes.xml:
```diff
@@ -165,7 +165,7 @@ public class Base {
 			</listitem>
 			<listitem>
-				<para>If you are using members of the type <classname>Nullables.NullableXXX</classname> (from the library <link linkend="nullables">Nullables</link>), then they will be mapped to <classname>Nullables.NHibernate.NullableXXXType</classname> automatically; don't set <literal>Type="..."</literal> in <classname>[Property]</classname> (leave it null). This is also the case for <classname>SqlTypes</ ...
+				<para>If you are using members of the type <classname>Nullables.NullableXXX</classname> (from the library Nullables), then they will be mapped to <classname>Nullables.NHibernate.NullableXXXType</classname> automatically; don't set <literal>Type="..."</literal> in <classname>[Property]</classname> (leave it null). This is also the case for <classname>SqlTypes</classname> (and you can add your  ...
```
I think this has to be removed completely. As this, probably does not happen currently. And, nobody is using Nullables library.

fredericDelaporte commented:
Sorry for the sandcastle project, I have removed it. I wanted to gitignore it, but done that in another branch. Now I have put  a .gitignore in this branch. The Nant build generates this sandcastle project, so it seems legit to ignore it in doc folder.
Hazzik, I have removed the obsolete paragraph.
